src/main/kotlin/com/craftschemers/hub/managers/HubMinigameManager.kt

```diff
@@ -1,3 +1,40 @@
 package com.craftschemers.hub.managers
 
-class HubMinigameManager
+import com.craftschemers.hub.HubPlayer
+import com.craftschemers.hub.minigame.GameState
+import com.craftschemers.hub.minigame.GameType
+import com.craftschemers.hub.minigame.Minigame
+import com.craftschemers.hub.minigame.OneInTheQuiver
+
+class HubMinigameManager {
+
+    val minigameServers = mutableMapOf<GameType, List<Minigame>>()
+
+    fun addPlayerToMinigame(player: HubPlayer, gameType: GameType) {
+        // make sure a server is available
+        val gameServer = minigameServers[gameType]
+        if (gameServer == null || gameServer.none { it.state == GameState.LOBBY }) {
+            // make a new server
+            val game = when (gameType) {
+                GameType.ONE_IN_THE_QUIVER -> OneInTheQuiver()
+            }
+            game.addPlayer(player)
+            minigameServers[gameType] = gameServer.orEmpty() + mutableListOf(game)
+        } else {
+            gameServer.find { it.state == GameState.LOBBY }?.addPlayer(player)
```

Comment:
Maybe we could extract `gameServer.find { it.state == GameState.LOBBY }` to a variable instead of having to write it twice?